client/client.go

```diff
@@ -426,18 +410,22 @@ func (c *Client) Watch(
 					Type: DocumentsChanged,
 					Keys: converter.FromDocumentKeys(resp.Event.DocumentKeys),
 				}, nil
-			case types.DocumentsWatchedEvent, types.DocumentsUnwatchedEvent:
+			case types.DocumentsWatchedEvent, types.DocumentsUnwatchedEvent, types.MetadataChangedEvent:
 				for _, k := range converter.FromDocumentKeys(resp.Event.DocumentKeys) {
 					cli, err := converter.FromClient(resp.Event.Publisher)
 					if err != nil {
 						return nil, err
 					}
 
 					attachment := c.attachments[k.BSONKey()]
-					if eventType == types.DocumentsWatchedEvent {
-						attachment.peerClients[cli.ID.String()] = cli.Metadata
+					if eventType == types.DocumentsWatchedEvent || eventType == types.MetadataChangedEvent {
+						if info, ok := attachment.peers[cli.ID.String()]; ok {
+							info.Update(cli.MetadataInfo)
```

Comment:
Currently, `attachment.peers` is of type `map[string]types.MetadataInfo`. `MetadataInfo` is not a pointer.
So even if you `Update()` the `info` variable, the instance managed inside `attachment.peers` is not updated. Is this intentional?

I think the WatchResponse is being returned as an unintended response value.
I think it would be nice to add a test that checks that the metadata of the Client instance and the peers information synced by other clients match.
Could tell me what you think?

Reply:
Thanks to you, we found a bug that would be hard to find later.